### PR TITLE
Update the `document` example to work

### PR DESCRIPTION
**Pull request by jobarr-amzn**
Tested with `ion-schema-cli validate --document [...]` using the CLI in `ion-schema-kotlin`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/isl-1-0/spec.md [hunk 1]
```diff
@@ -1191,32 +1191,39 @@ The following schema provides an example of using the 'document' type,
...
+$ion_schema_1_0
+
 schema_header::{
 }
 
 type::{
   name: IonSchema,
   type: document,
   ordered_elements: [
+    { type: SchemaVersionMarker, occurs: required },
```
Comment (popematt) [inline docs/isl-1-0/spec.md]: Schema version marker is not actually required for Ion Schema 1.0.
```suggestion
    { type: SchemaVersionMarker, occurs: optional },
```

File: docs/isl-1-0/spec.md [hunk 2]
```diff
@@ -1191,32 +1191,39 @@ The following schema provides an example of using the 'document' type,
...
     { type: Header, occurs: optional },
     { type: Type,   occurs: range::[0, max] },
     { type: Footer, occurs: optional },
   ],
 }
 
+type::{
+  name: SchemaVersionMarker,
+  valid_values: [$ion_schema_1_0, $ion_schema_2_0]
```
Comment (popematt) [inline docs/isl-1-0/spec.md]: Seems a little weird to me for the ISL 1.0 spec have a reference to ISL 2.0. What do you think?

File: docs/isl-1-0/spec.md [hunk 3]
```diff
@@ -1225,7 +1232,6 @@ type::{
   annotations: [required::type],
   fields: {
     name: symbol,
-    ...
```
Comment (popematt) [inline docs/isl-1-0/spec.md:1228]: What about changing it to a comment?
```
// ...
```
I think that helps show that it is not intended to be a complete definition.

File: docs/isl-1-0/spec.md (hunk 2 above)
Comment (jobarr-amzn) [reply docs/isl-1-0/spec.md]: That's a good point, will remove.

File: docs/isl-1-0/spec.md (hunk 3 above)
Comment (jobarr-amzn) [reply docs/isl-1-0/spec.md:1228]: Excellent suggestion, I should have done that. Will do!

**Review by popematt — APPROVED** [review]

**Pull request merged**